src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -673,3 +676,328 @@ The Run handle/object supports few other use-cases.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The API ``xrtGraphTimeStamp`` can be used to determine AIE cycle consumed between a graph start and stop. 
+
+Here in this example, the AIE cycle consumed by 3 iteration is calculated
+ 
+
+.. code:: c++
+      :number-lines: 35
+           
+           // start from reset state
+           xrtGraphReset(graphHandle);
+           
+           uint64_t begin_t = xrtGraphTimeStamp(graphHandle);
+           
+           // run the graph for 3 iteration
+           xrtGraphRun(graphHandle, 3);
+           
+           xrtGraphWait(graphHandle, 0); 
+           
+           uint64_t end_t = xrtGraphTimeStamp(graphHandle);
+           
+           std::cout<<"Number of AIE cycles consumed in the 3 iteration is: "<< end_t-begin_t; 
+           
+
+RTP (Runtime Parameter) control
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+XRT provides the API to update and read the runtime parameters of the graph. 
+
+- The API ``xrtGraphUpdateRTP`` to update the RTP 
+- The API ``xrtGraphReadRTP`` to read the RTP. 
+
+.. code:: c++
+      :number-lines: 35
+
+           ret = xrtGraphReset(graphHandle);
+           if (ret) throw std::runtime_error("Unable to reset graph");
+
+           ret = xrtGraphRun(graphHandle, 2);
+           if (ret) throw std::runtime_error("Unable to run graph");
+
+           float increment[1] = {1};
+           const char *inVect = reinterpret_cast<const char *>(increment);
+           xrtGraphUpdateRTP(graphHandle, "mm.mm0.in[2]", inVect, sizeof (float));
+     
+           // Do more things
+           xrtGraphRun(graphHandle,16);
+           xrtGraphWait(graphHandle,0);
+     
+           // Read RTP
+           float increment_out[1] = {1};
+           char *outVect = reinterpret_cast<char *>(increment_out);
+           xrtGraphReadRTP(graphHandle, "mm.mm0.inout[0]", outVect, sizeof(float));
+           std::cout<<"\n RTP value read<<increment_out[0]; 
+ 
+In the above example, the API ``xrtGraphUpdateRTP`` and ``xrtGraphReadRTP`` are used to update and read the RTP values respectively. Note the API arguments 
+   
+      - The hierarchical name of the RTP port
+      - Pointer to write or read the RTP variable
+      - The size of the RTP value. 
+
+DMA operation to and from Global Memory IO
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+XRT provides API ``xrtAIESyncBO`` to synchronize the buffer contents between GMIO and AIE. The following code shows a sample example
+
+
+.. code:: c++
+      :number-lines: 35
+
+           xrtDeviceHandle device_handle = xrtDeviceOpen(0);
+       
+           // Buffer from GM to AIE
+           xrtBufferHandle in_bo_handle  = xrtBOAlloc(device_handle, SIZE * sizeof (float), 0, 0);
+       
+           // Buffer from AIE to GM
+           xrtBufferHandle out_bo_handle  = xrtBOAlloc(device_handle, SIZE * sizeof (float), 0, 0);
+       
+           inp_bo_map = (float *)xrtBOMap(in_bo_handle);
+           out_bo_map = (float *)xrtBOMap(out_bo_handle);
```

Comment:
```suggestion
           auto out_bo_map = out_bo_handle.map<float>();
```
Just look at SYCL if you are looking for modern C++ stylistic inspiration... :-)